0018-connector-risk-mitigations/0018-connector-risk-mitigations.md

```diff
@@ -16,29 +14,70 @@ The main risk connectors face in Interledger is being unable to fulfill the inco
-* **Preferring Smaller Payments** - Smaller payments place a smaller percentage of the connector's liquidity on hold and thus each one carries less risk that the payment being prepared and then failing would create significant opportunity cost
-* **Blacklisting Senders and Receivers** - Connectors may refuse to facilitate payments from certain sources or to certain destinations if their rate of failed payments is unusually high
-
-## Exchange Rate Volatility
+* **Preferring Smaller Payments** - Smaller payments place a smaller percentage of the connector's liquidity on hold and thus each one carries less risk that the payment being prepared and then failing would create significant opportunity cost - in ILPv4, this is the default.
+* **Blacklisting Senders and Receivers** - Connectors may refuse to facilitate payments from certain sources or to certain destinations if their rate of failed payments is unusually high.
 
-Once connectors prepare their outgoing transfers, they are committed to the payment even if the exchange rate between the assets fluctuates.
-
-Possible approaches include:
-
-* **Accounting for Slippage** - Connectors may add a buffer to their expected exchange rate to account for some movement in the price, and they may add a premium for especially volatile currencies
-* **Preferring Shorter Timeouts** - Connectors may provide better exchange rates for payments with shorter timeouts because they carry less risk that the price will move drastically while the payment is in flight
+Counterparty Theft: Suppose you have connectors A, B, C and D, and the payment is going from A->B->C->D. Take the case of C. During the fulfill phase, D will pass a fulfill packet to C, meaning that C will have to pay. But because ILP is using unconditional payment channels, C can write a malicious client that could forward a fulfill packet to B, and avoid paying D. In this case, A would pay B, B would pay C, but C didn't pay D.
 
+Possible mitigations include:
+* **Accept only smaller payments** - If the payment per packet is less than that
+    of an on chain-fee, a connector that cheats its counterparty would lose much
+    more capital in the process. For example, a connector that has been deprived of
+    payment from its counterparty could refuse to facilitate future payments,
+    and the malicious counterparty would have to close their payment channel to
+    make the stolen micropayment amount liquid. Given that on-chain transaction fees
+    are much higher than that of a micropayment in Interledger (i.e. 1 satoshi), an
+    attacker would lose more money open and closing channels to cheat their
+    counterparty. Therefore, while this griefing attack is possible, if
+    connectors are leveraging smaller payments, the attacking connector is more incentivized
+    to facilitate payments to make profit on the exchange rate/fees.
+* **Provide fraud proof support on the ledger** - At the moment, ILP does not
```

Comment:
> I think fraud proofs, as they are mentioned here, are out of scope for this RFC, but it could lead to an interesting new proposal if you have a good idea for how it could be constructed.

Yea I'll move it into another proposal. I think this may only work in certain cases and I don't expect most chains to provide endpoints to do it. I'll write up a proposal and publish if it makes sense. Thanks!